src/components/ViewedByCustomer/ViewedByCustomer.ts

```diff
@@ -0,0 +1,65 @@
+import { Component, ComponentOptions, IComponentBindings, $$, Initialization } from 'coveo-search-ui';
+import { user } from '../../utils/icons';
+
+/**
+ * The options for the ViewedByCustomerComponent
+ */
+export interface IViewedByCustomerOptions {
+    showIcon?: boolean;
+    label?: string;
+}
+
+/**
+ * Result component indicating if a search result have been viewed by the targeted user.
+ */
+export class ViewedByCustomer extends Component {
+    /**
+     * Unique Identifier used by the Search-UI.
+     */
+    public static readonly ID = 'ViewedByCustomer';
+
+    /**
+     * Default options used by the component.
+     */
+    public static readonly options: IViewedByCustomerOptions = {
+        /**
+         * If true, will display an icon when the component is displayed itself.
+         */
+        showIcon: ComponentOptions.buildBooleanOption({ defaultValue: true }),
+        /**
+         * The label that should be displayed when the component is displayed.
+         */
+        label: ComponentOptions.buildStringOption({ defaultValue: 'Viewed by Customer' })
+    };
+
+    /**
+     * Create an instance of {@link ViewedByCustomer}.
+     * @param element Element on which to bind the component.
+     * @param options Initialization options of the component.
+     * @param bindings Bindings of the Search-UI environment.
+     */
+    public constructor(public element: HTMLElement, public options: IViewedByCustomerOptions, public bindings: IComponentBindings) {
+        super(element, ViewedByCustomer.ID, bindings)/* istanbul ignore next Issue with Istanbul and super calls*/;
+        this.options = ComponentOptions.initComponentOptions(element, ViewedByCustomer, options);
+
+        if (this.resolveResult().isUserActionView) {

```

Comment:
Where is this `isUserActionView` coming from? 
Is it from the generic "Component" from the jsui?

I remember we talked at some point about potentially greying out the result when it had been already viewed, any opinion on adding a potential CSS class to the entire result rendered DOM when it is visited?

Reply:
It's from `IQueryResult` of the coveo/search-ui. See coveo/search-ui#1170

Reply:
So what about the answer to my other question?

Reply:
About your other question, as you said this was talked about but never specified (in the 'specification' meaning), thus, not implemented there.

More importantly, this is the bare-MVP for ViewedByCustomer, I have no objection to improve it later, but, well, _later_.